webclient/dokument/views.py

```diff
@@ -1038,21 +1040,30 @@ def get_context_data(self, **kwargs):
 
     def post(self, request, *args, **kwargs):
         cast = self.get_zaznam()
-        cast.create_transaction(request.user, self.success_message)
-        cast.close_active_transaction_when_finished = True
+        fedora_transaction = cast.create_transaction(request.user, self.success_message)
         archeologicky_zaznam = cast.archeologicky_zaznam
         cast.archeologicky_zaznam = None
         cast.projekt = None
-        cast.save()
-        if (
-            archeologicky_zaznam
-            and archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
-            and archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
-            and archeologicky_zaznam.lokalita.igsn
-        ):
-            archeologicky_zaznam.lokalita.igsn_update()
-        if cast.dokument.stav == D_STAV_ARCHIVOVANY and cast.dokument.doi:
-            cast.dokument.doi_update()
+        try:
+            cast.save()
+            if (
+                archeologicky_zaznam
+                and archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
+                and archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
+                and archeologicky_zaznam.lokalita.igsn
+            ):
+                archeologicky_zaznam.lokalita.igsn_update()
+            if cast.dokument.stav == D_STAV_ARCHIVOVANY and cast.dokument.doi:
+                cast.dokument.doi_update()
+            cast.close_active_transaction_when_finished = True
+            cast.save()
```

Comment:
Proč se cast.save volá dvakrát? (na začátku zůstat musí, ale proč je i tady na konci?

Reply:
Na konci to je kvůli uzavření transakce ve Fedoře.